order.go

```diff
@@ -262,6 +262,30 @@ type ShippingLines struct {
 	TaxLines                      []TaxLine        `json:"tax_lines,omitempty"`
 }
 
+// UnmarshalJSON custom unmarshaller for ShippingLines implemented to handle requested_fulfillment_service_id being
+// returned as json numbers or json nulls instead of json strings
+func (sl *ShippingLines) UnmarshalJSON(data []byte) error {
+	type alias ShippingLines
```

Comment:
Is it necessary to use alias here? It looks like you can unmarshal it to the structure with just one field you need.  If alias is still needed, please could you give more meaningful name?

Reply:
The purpose of `alias` in this code is to override `RequestedFulfillmentServiceID` in the `ShippingLines` struct, that way we can deserialize the rest of the fields into `sl` and `RequestedFulfillmentServiceID` into `aux` as `interface{}`.

If we unmarshalled into a struct with a single field we'd have to unmarshal twice and ignore the error we'd getting from unmarshalling something a json number like `123` for instance into `sl.RequestedFulfillmentServiceID` which is a string.

In regards to the name of the struct `alias`, knowing that I'm the worst when it comes to naming things 😆 I decided to follow the convention stablished here:
https://github.com/bold-commerce/go-shopify/blob/0398b1e978c09b00d9dfba2d37a5442b9865bb7f/order.go#L190-L195

But that's super easy to change, what name do you suggest?

Reply:
Oh, I did not see that we use `alias` somewhere else in code. In this case I think we keep it as is to be consistent. 

I am wondering if we actually need to use original structure and override the field in it. Usually in this case I just create a new structure with just one field I need and I believe JSON unmarshaler will just ignore all other fields. However if current approach allow to avoid calling unmarshaler twice, it is better approach than I suggesting